tests/unit/processor/hyperscan_resolver/test_hyperscan_resolver.py

```diff
@@ -0,0 +1,769 @@
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_from_file": "tests/testdata/unit/hyperscan_resolver/resolve_mapping.yml",
+                "resolve_list": {"FOO": "BAR"},
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {
+            "to_resolve": "not_in_list",
+        }
+        document = {"to_resolve": "not_in_list"}
+
+        self.object.process(document)
+
+        assert document == expected
+
+    def test_resolve_dotted_no_conflict_from_file_and_list(
+        self,
+    ):
+        rule = {
+            "filter": "to_resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_from_file": "tests/testdata/unit/hyperscan_resolver/resolve_mapping.yml",
+                "append_to_list": True,
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {"to_resolve": "12ab34", "resolved": ["ab_server_type"]}
+        document = {"to_resolve": "12ab34"}
+
+        self.object.process(document)
+
+        assert document == expected
+
+    def test_resolve_dotted_no_conflict_from_file_and_list_has_conflict(
+        self,
+    ):
+        rule = {
+            "filter": "to_resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_from_file": "tests/testdata/unit/hyperscan_resolver/resolve_mapping.yml",
+                "append_to_list": True,
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {"to_resolve": "12ab34", "resolved": ["ab_server_type"]}
+        document = {"to_resolve": "12ab34"}
+
+        self.object.process(document)
+        self.object.process(document)
+
+        assert document == expected
+
+    def test_resolve_dotted_no_conflict_from_file_and_list_has_conflict_and_diff_inputs(
+        self,
+    ):
+        rule = {
+            "filter": "to_resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to_resolve": "resolved", "other_to_resolve": "resolved"},
+                "resolve_from_file": "tests/testdata/unit/hyperscan_resolver/resolve_mapping.yml",
+                "append_to_list": True,
+            },
+        }
+
+        self._load_specific_rule(rule)
+
+        expected = {
+            "to_resolve": "12ab34",
+            "other_to_resolve": "00de11",
+            "resolved": ["ab_server_type", "de_server_type"],
+        }
+        document = {"to_resolve": "12ab34", "other_to_resolve": "00de11"}
+
+        self.object.process(document)
+        self.object.process(document)
+
+        assert document == expected
+
+    def test_resolve_from_file_and_file_does_not_exist(self):
+        rule = {
+            "filter": "to.resolve",
+            "hyperscan_resolver": {
+                "field_mapping": {"to.resolve": "resolved"},
+                "resolve_from_file": "foo",
```

Comment:
```suggestion
                "resolve_from_file": "i/do/not/exist",
```
make it more pathlike and catchy

Reply:
I changed it.